pkg/fetcher/fetcher_archive.go

```diff
@@ -0,0 +1,150 @@
+package fetcher
+
+import (
+	"errors"
+	"path"
+	"strings"
+
+	"github.com/readium/go-toolkit/pkg/archive"
+	"github.com/readium/go-toolkit/pkg/mediatype"
+	"github.com/readium/go-toolkit/pkg/pub"
+)
+
+// Provides access to entries of an archive.
+type ArchiveFetcher struct {
+	archive archive.Archive
+}
+
+func (f *ArchiveFetcher) Links() ([]pub.Link, error) {
+	entries := f.archive.Entries()
+	links := make([]pub.Link, 0, len(entries))
+	for _, af := range entries {
+		fp := path.Clean(af.Path())
+		if !strings.HasPrefix(fp, "/") {
+			fp = "/" + fp
+		}
+		link := pub.Link{
+			Href: fp,
+		}
+		ext := path.Ext(fp)
+		if ext != "" {
+			mt := mediatype.MediaTypeOfExtension(ext[1:]) // Remove leading "."
+			if mt != nil {
+				link.Type = mt.String()
+			}
+		}
+		cl := af.CompressedLength()
+		if cl == 0 {
+			cl = af.Length()
+		}
+		link.Properties.Add(pub.Properties{
+			"archive": pub.Properties{
+				"entryLength":       cl,
+				"isEntryCompressed": af.CompressedLength() > 0,
+			},
+		})
+		links = append(links, link)
+	}
+	return links, nil
+}
+
+func (f *ArchiveFetcher) Get(link pub.Link) Resource {
+	entry, err := f.archive.Entry(strings.TrimPrefix(link.Href, "/"))
+	if err != nil {
+		return NewFailureResource(link, NotFound(err))
+	}
+	return &entryResource{
+		link:  link,
+		entry: entry,
+	}
+}
+
+func (f *ArchiveFetcher) Close() {
+	f.archive.Close()
+}
+
+func NewArchiveFetcher(a archive.Archive) *ArchiveFetcher {
+	return &ArchiveFetcher{
+		archive: a,
+	}
+}
+
+func NewArchiveFetcherFromPath(filepath string) (*ArchiveFetcher, error) {
+	archive, err := archive.NewArchiveFactory().Open(filepath, "")
+	if err != nil {
+		return nil, err
+	}
+	return &ArchiveFetcher{
+		archive: archive,
+	}, nil
+}
+
+func NewArchiveFetcherFromPathWithFactory(path string, factory archive.ArchiveFactory) (*ArchiveFetcher, error) {
+	a, err := factory.Open(path, "") // TODO password
+	if err != nil {
+		return nil, err
+	}
+	return &ArchiveFetcher{
+		archive: *a,
+	}, nil
+}
+
+// Resource from archive entry
+type entryResource struct {
+	link  pub.Link
+	entry archive.Entry
+}
+
+func (r *entryResource) File() string {
+	return ""
+}
+
+func (r *entryResource) Close() {
+	// Nothing needs to be done at the moment
+}
+
+func (r *entryResource) Link() pub.Link {
+	cl := r.entry.CompressedLength()
+	if cl == 0 {
+		cl = r.entry.Length()
+	}
+	r.link.Properties.Add(pub.Properties{
+		"archive": pub.Properties{
```

Comment:
I suggest using the `https://readium.org/webpub-manifest/properties#` prefix for internal properties. We can always add a section on this page to add some explanations for each internal property.

```suggestion
		"https://readium.org/webpub-manifest/properties#archive": pub.Properties{
```